shell/browser/api/atom_api_url_request.cc

```diff
@@ -554,6 +571,11 @@ void URLRequest::EmitError(EventType type, base::StringPiece message) {
   EmitEvent(type, false, "error", error);
 }
 
+void URLRequest::EmitFinished() {
+  request_state_ |= STATE_FINISHED;
```

Comment:
can we `DCHECK` that we didn't already emit this event?

Reply:
updated to emit once try to follow https://github.com/electron/electron/pull/21157/files#diff-83b26000afb8e61c607f454ace5a06d2R248-R249, but if DCHECK makes more sense will update again.

Reply:
@kwonoj If finishing twice is unexpected / invalid we should `DCHECK` on it to ensure we notice in the tests.  If it's an unnaceptable state we should `CHECK` on it.  Otherwise if it's just a state we can ignore then the current code is fine.  Personally I think it should be some form of check

Reply:
I think trying to finish multiple time in URLRequest itself is fine, as long as consumer gauranteed to have single finish event before close.